primeqa/components/base.py

```diff
@@ -155,3 +155,44 @@ def get_engine_type(self) -> str:
     @abstractmethod
     def index(self, collection: Union[List[dict], str], *args, **kwargs):
         pass
+    
+@dataclass(init=False, repr=False, eq=False)
+class Reranker(Component):
+    
+    @abstractmethod
+    def __hash__(self) -> int:
+        """
+        Custom hashing function useful to compare instances of `Retriever`.
+
+        Raises:
+            NotImplementedError:
+
+        Returns:
+            int: hash value
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def predict(self, queries: List[str], 
+                    documents: List[List[Dict]],
+                    *args, 
+                    **kwargs):
+        """
+        Args:
+            queries (List[str]): search queries
+            texts (List[List[Dict]]): For each query, a list of documents to rerank
+                where each document is a dictionary with the following structure:
+                {
+                    "document": {
+                        "text": "A man is eating food.",
+                        "document_id": "0",
+                        "title": "food"
+                    },
+                    "score": 1.4
```

Comment:
do we necessarily need the score parameter to be present in the doc dict ? Are we using it in our re-ranker ?